File: cpp/easy/FindSpecialInteger.cpp
#include <vector>

using namespace std;

int findSpecialInteger(vector<int> &arr) {
  int target = arr.size() / 4;
  int count = 1;
  int rs = arr[0];
  for (int i = 1; i < arr.size(); ++i) {
    if (arr[i] == arr[i - 1]) {
      count++;
    } else {
      count = 1;
      rs = arr[i];
    }
    if (count > target) {
      return rs;
    }
  }
  return rs;
}

File: typescript/easy/findSpecialInteger.ts
function findSpecialInteger(arr: number[]): number {
  let sum: number = 0;
  for (let i = 0; i < arr.length; i++) {
    sum += arr[i];
  }
  let left: number = 0;
  let right: number = arr.length - 1;
  while (left <= right) {
    const mid: number = left + (right - left) / 2;
  }
}
